sygnal/sygnal.py

```diff
@@ -52,18 +63,33 @@
 }
 
 
+class SygnalReactor(
+    IReactorFDSet,
+    IReactorPluggableNameResolver,
+    IReactorTCP,
+    IReactorCore,
+    Interface,
+):
+    pass
+
+
 class Sygnal:
-    def __init__(self, config, custom_reactor, tracer=opentracing.tracer):
+    def __init__(
+        self,
+        config: Dict[str, Any],
+        custom_reactor: SygnalReactor,
+        tracer: Union[Tracer, jaeger_client.Tracer] = opentracing.tracer,
```

Comment:
I think you should be able to just use `Tracer`, because [jaeger_client.Tracer inherits from opentracing.Tracer](https://github.com/jaegertracing/jaeger-client-python/blob/a81052e73d238692ed5c6101aab420f51d414a64/jaeger_client/tracer.py#L46).